common/src/main/java/io/druid/timeline/TimelineLookup.java

```diff
@@ -35,6 +35,8 @@
    */
   public Iterable<TimelineObjectHolder<VersionType, ObjectType>> lookup(Interval interval);
 
+  public Iterable<TimelineObjectHolder<VersionType, ObjectType>> lookup(Interval interval, boolean incompleteOk);
```

Comment:
Can we instead expose the method

```
public Iterable<TimelineObjectHolder<VersionType, ObjectType>> lookupWithIncomplete(Interval interval);
```

Boolean flag style method often result in pretty confusing function signatures, because you are not always sure of what the boolean is supposed to mean.  Especially if you don't actually have the code.  If we can instead move those semantics into a completely different method name, it becomes easier for users to know what is going on.

Also, :+1: on adding javadoc.